MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -0,0 +1,660 @@
+import Foundation
+import CoreLocation
+import MapboxDirections
+import Polyline
+import MapboxMobileEvents
+import Turf
+
+
+protocol RouteControllerDataSource: class {
+    var location: CLLocation? { get }
+    var locationProvider: NavigationLocationManager.Type { get }
+}
+
+
+/**
+ A `RouteController` tracks the user’s progress along a route, posting notifications as the user reaches significant points along the route. On every location update, the route controller evaluates the user’s location, determining whether the user remains on the route. If not, the route controller calculates a new route.
```

Comment:
This documentation refers to RouteController but documents LegacyRouteController. Note that jazzy lacks support for deprecation warnings: realm/jazzy#843. In the past, we’ve gone ahead and removed documentation comments from deprecated symbols, to avoid promoting their continued usage. However, this has historically been such an important class that we might want to leave the comment in place and add a manual deprecation note.